addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,634 @@
+                else:
+                    sinvoice_status = 'paid'
+
+            if invoice.payment_state in {'in_payment', 'paid'} and sinvoice_status == 'unpaid':
+                # Mark the invoice as paid
+                endpoint = 'InvoiceAPI/InvoiceWS/updatePaymentStatus'
+                params = {
+                    'supplierTaxCode': invoice.company_id.vat,
+                    'templateCode': invoice.l10n_vn_edi_invoice_symbol.invoice_template_id.name,
+                    'invoiceNo': invoice.l10n_vn_edi_invoice_number,
+                    'strIssueDate': invoice._l10n_vn_edi_format_date(invoice.l10n_vn_edi_issue_date),
+                }
+            elif invoice.payment_state not in {'in_payment', 'paid'} and sinvoice_status == 'paid':
+                # Mark the invoice as not paid
+                endpoint = 'InvoiceAPI/InvoiceWS/cancelPaymentStatus'
+                params = {
+                    'supplierTaxCode': invoice.company_id.vat,
+                    'invoiceNo': invoice.l10n_vn_edi_invoice_number,
+                    'strIssueDate': invoice._l10n_vn_edi_format_date(invoice.l10n_vn_edi_issue_date),
+                }
+
+            if endpoint and params:
+                request_response, error_message = invoice._l10n_vn_edi_send_request(
+                    method='POST',
+                    endpoint=endpoint,
+                    params=params,
+                )
+
+                if error_message:
+                    raise UserError(error_message)
+
+                # Revert back to the sent state as the status is up-to-date.
+                invoice.l10n_vn_edi_state = 'sent'
+
+    def _need_cancel_request(self):
+        # EXTEND 'account'
+        return super()._need_cancel_request() or self.l10n_vn_edi_invoice_number
+
+    def _post(self, soft=True):
+        # EXTEND 'account'
+        posted = super()._post(soft)
+
+        # Ensure to tag the move as 'Ready to send' upon posting if it makes sense.
+        for invoice in posted.filtered(lambda i: not i.l10n_vn_edi_state):
+            if invoice.company_id.account_fiscal_country_id.code == 'VN' and invoice.is_sale_document() and not invoice.l10n_vn_edi_invoice_number:
+                invoice.l10n_vn_edi_state = 'ready_to_send'
+
+        return posted
+
+    # API
+
+    def _l10n_vn_edi_check_invoice_configuration(self):
+        """ Some checks that are used to avoid common errors before sending the invoice. """
+        self.ensure_one()
+        company = self.company_id
+        commercial_partner = self.commercial_partner_id
+        errors = []
+        if not company.l10n_vn_edi_username or not company.l10n_vn_edi_password:
+            errors.append(_('Sinvoice credentials are missing on company %s.', company.display_name))
+        if not company.vat:
+            errors.append(_('VAT number is missing on company %s.', company.display_name))
+        if company.phone and not company.phone.replace(' ', '').isdecimal():
```

Comment:
Probably need to check the `seller address`, `street` field maybe?

Reply:
I added a check on the street, city, state, and country (buyer) and street, state, country (seller) as these are asked by sinvoice. I believe SInvoice won't complain if some are missing, but I believe these are required to have the complete and corrct address of the user :thinking: 

Thanks

Reply:
Indeed some address info is required for the seller and i check again with documentation almost all `buyer` info is not required at all but i agree we will need them not blank :v